File: library/src/main/java/com/xingyun/library/utils/EvenLog.java
package com.xingyun.library.utils;

import android.util.Log;

public class EvenLog {
	private static final String TAG = "EvenLog";

	public static void d(String message) {
		d(TAG, message);
	}

	public static void d(String tag, String message) {
		Log.d(tag, message);
	}

}

File: app/src/main/java/com/xingyun/evendemo/utils/EvenLog.kt
package com.xingyun.evendemo.utils

import android.util.Log
import com.google.gson.Gson

class EvenLog {

    companion object {
        private const val TAG = "EvenChen"

        fun d(message: String) {
            Log.d(TAG, message)
        }

        fun d(position: String, message: String) {
            d("$position -> $message")
        }

    }

}

inline fun <reified T : Any> EvenLog.d(position: String, bean: T) {
    val toString = Gson().toJson(bean)
    EvenLog.d("$position -> $toString")
}